### DISCUSSION TITLE
Death penalty

### DISCUSSION THREAD

**Discussion by Dutasmilas** [Zrips/CMI#5946] [Feature Requests]
I would like this feature to be implemented for CMI

Player dying will lose certain percentage or certain amount of money they had.

so server with keep inventory on will have more balancing economy.

**Comment by CloudeLecaw** [comment]
You can do this allready by simply using ancommand in EventCommands when they die

**Comment by Dutasmilas** [comment]
But I would like a feature that take certain percentage amount that they had,they aren’t any of this option.

for example if they had 1mil,10% would be 1000000 of that.So it not a fix amount when player die.

**Comment by CloudeLecaw** [comment]
> But I would like a feature that take certain percentage amount that they had,they aren’t any of this option.
> 
> for example if they had 1mil,10% would be 1000000 of that.So it not a fix amount when player die.

this is not correct.. take alook in the wiki...Command structure like seen in wiki is:

> money [pay/give/take/set] [playerName/all/alloffline] [amount/%rand/1-1000%/1%[min-max][playerName]] (-s)

Open eventCommands.yml

```
playerDeath:
  Enabled: true
  Commands:
  - msg! [playerName] &cYou lost 10% of Your current balance.
  - cmi money take [playerName] 10% -s
```

Now when a player dies he loose 10% of all he has

You see you can also define condistion like loose 10% but maximum 50000 or 10% but minimum 1000 or something like that...

![image](https://user-images.githubusercontent.com/12080017/126109419-826cc2ab-d2cd-493a-b435-89dffc72903d.png)
here u see
- i checked my balance
- i used take 10% of my money  (-s to have no meassage like server took x money)
- i checked balance aggain

**Comment by CloudeLecaw** [comment]
Well u Bring me to the idea to suggest this for exp too :)
Currently this is not possible for xp.

**Comment by Dutasmilas** [comment]
How about cooldown for it?So player can’t abuse killing each other and they keep losing money

**Comment by CloudeLecaw** [comment]
Well why players should do this...
If they kill each other each time... They loose money
Whats a abuse of not having money?
You could define an imortal time after they respawned in config to prevent instand pks.
But a cooldown for this does not exist at the moment... Well thats a new Suggestion Feature and wasnt suggest at the beginning of your thread :3

May something could be added like
A placeholder that can be used to check if the death was a playerkill or not..but this would just be abused too with simply killing the other before he dies normaly or by monsters

May there is no perfect solution for this Situation..
Players allways find ways to abuse a System..and after that..they call it feature..

**Comment by Dutasmilas** [comment]
Okay how about if a player more than certain amount of money then they get money deduct if they die.
How do I put ?

**Comment by CloudeLecaw** [comment]
you mean only execute if player has more than x money? simply add that check before the commands

```
playerDeath:
  Enabled: true
  Commands:
  - check:%cmi_user_balance%>100! msg! [playerName] &cYou lost 10% of Your current balance.
  - check:%cmi_user_balance%>100! cmi money take [playerName] 10% -s
```

With this extra check the commands only would executed if the Players money is over 100

Note about checks:
```
If command start with check:[value1][==|>|>=|<|<=|!=][value2][@][?][#]! then we will only proceed if condition is correct.
You also can check for equals and unquals of words or any other things
? means inform player - well can be used if a permission is asked for and he should get informed that he does not has it..
# means cannel all after this if the requiement is not filled
@ means do it only if the requirement is not true

```
[More Information about cmis specialized commands you find here:](https://www.zrips.net/cmi/commands/specialized/)
